src/device-api/v2.ts

```diff
@@ -424,25 +424,34 @@ router.get('/v2/containerId', async (req: AuthorizedRequest, res) => {
 router.get('/v2/state/status', async (req: AuthorizedRequest, res) => {
 	const appIds: number[] = [];
 	const pending = deviceState.isApplyInProgress();
+	const allImages = await images.getState();
 	const containerStates = (await serviceManager.getAll())
 		.filter((service) => req.auth.isScoped({ apps: [service.appId] }))
 		.map((svc) => {
 			appIds.push(svc.appId);
-			return _.pick(
+			const s = _.pick(
 				svc,
 				'status',
 				'serviceName',
 				'appId',
-				'imageId',
 				'serviceId',
+				'imageId',
 				'containerId',
 				'createdAt',
 			);
+
+			// QUESTION: what to do if we cannot find the image?
+			// if the image database got deleted
+			s.imageId = allImages.find(
+				(img) => img.commit === svc.commit && img.serviceName === s.serviceName,
+			)?.imageId;
+
+			return s;
```

Comment:
I don't think we need to worry about this, if the image table got deleted unexpectedly, then the action would have thrown a lot earlier during `images.getState`

Reply:
I don't think `images.getState` would throw, but we recreate the image database when initializing application manager so if the data from `getState` should be correct, so you are right the comment is unnecessary